src/openmrs-esm-module-config.ts

```diff
@@ -0,0 +1,49 @@
+import * as R from "ramda";
+
+// The configurations that have been provided
+const configs : object[] = [];
+
+// An object with module names for keys and schemas for values.
+const schemas = {};
+
+export function defineConfigSchema(moduleName, schema) {
+  console.log("defineConfigSchema received schema for " + moduleName + ": " + JSON.stringify(schema));
+  schemas[moduleName] = schema;
+}
+
+export function provide(config) {
+  console.log("provide recieved config " + JSON.stringify(config));
+  configs.push(config);
+}
+
+export function getConfig(moduleName) {
+  if (!schemas.hasOwnProperty(moduleName)) {
+    throw Error("No config schema has been defined for " + moduleName);
+  }
+  const schema = schemas[moduleName];
+
+  // Create a config object composed of all the defaults
+  const defaultConfig = R.map(R.prop('default'), schema);
+
+  // Merge all of the configs provided for moduleName
+  const allConfigsForModule = R.map(R.prop(moduleName), configs)
+  const providedConfig = R.mergeAll(allConfigsForModule);
```

Comment:
Looks like mergeAll enforces "last one wins"? Meaning the rightmost config in the array will override previous configs.

I'm assuming this is intended -- the last config you provide has more priority than the first?

Reply:
Yep, exactly